Merge mainline

vkublytskyi opened the pull request:
Merge mainline for valid PRs verification

driskell committed:
Magento PayPal checkout fails if email sending fails / other payment method does not

Commit:
made attributes nillable so we can reset parent settings

Commit:
parse nil values correctly

dmytro-ch committed:
Fix typo in DeploymentConfig class

jissereitsma committed:
Enhancements to module:status command

jissereitsma committed:
Add strict-modes

jissereitsma committed:
Add RETURN_FAILURED return if things fail

sanganinamrata committed:
Correct spelling mistakes in Model and library files.

Commit:
made more attributes nillable

yogeshsuhagiya committed:
Translated validation error messages

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento-engcom/msi?pullRequest=1568) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento-engcom/msi?pullRequest=1568) before we can accept your contribution.<br/>**1** out of **11** committers have signed the CLA.<br/><br/>:white_check_mark: naydav<br/>:x: joanhe<br/>:x: olysenko<br/>:x: magento-cicd2<br/>:x: magento-engcom-team<br/>:x: sidolov<br/>:x: zakdma<br/>:x: serhii-balko<br/>:x: nikshostko<br/>:x: thomas-blackbird<br/>:x: vkublytskyi

iGerchak committed:
Fix sending duplicate emails

GraysonChiang committed:
Fix Custom Attribute Group can not translation in catalog/product page

Commit:
Fix failed test EmailSenderHandlerTest

hostep committed:
Use '.min' in filenames of already minified js files in the Swagger module so they aren't getting minified again in production, fixes #16927 - for Magento 2.3

Commit:
Merge remote-tracking branch 'origin/2.3-develop' into assert-response-field-refactor

Commit:
Merge remote-tracking branch 'origin/MAGETWO-90719-reorder-exception' into BugFixPR

Commit:
Merge remote-tracking branch 'origin/MAGETWO-90632' into BugFixPR

Commit:
Merge remote-tracking branch 'origin/MAGETWO-90862-paypal-link' into BugFixPR

rogyar committed:
Added unit test for newsletter problem model

rogyar committed:
Minor naming adjustments

Commit:
Merge pull request #3023 from magento-chaika/chaika_august2

[Chaika] bugfixes

Commit:
merge magento/2.3-develop into magento-borg/BugFixPR

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.3-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
ENGCOM-2777: [Forwardport] Catalog: Add unit tests for Cron classes #17572

 - Merge Pull Request magento/magento2#17572 from jignesh-baldha/magento2:2.3-develop-PR-port-17561
 - Merged commits:
   1. 50e74cd04b835760ea5b53cad47f61ec5f83892a
   2. 077bf2bba31d4f3236fba6fd8693245cd7be87a4

Commit:
Merge remote-tracking branch 'origin/ENGCOM-2777-magento-magento2-17572' into 2.3-develop-prs

Commit:
Fixed code style issue

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into 2.3-develop-pr32

zakdma committed:
Merge remote-tracking branch 'origin/MAGETWO-60377' into 2.3-develop-pr32

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-93963' into 2.3-develop-pr1

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-73774' into 2.3-develop-pr1

nikshostko committed:
Merge pull request #53 from magento-epam/MAGETWO-66666

MAGETWO-66666: Adding a product to cart from category page with an ex…

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-93286' into 2.3-develop-pr1

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-93692' into 2.3-develop-pr1

nikshostko committed:
Merge pull request #54 from magento-epam/2.3-develop

Merge 2.3-develop to EPAM-PR-2

nikshostko committed:
MAGETWO-91620: It is not possible to use function setInAllAttributeSetsFilter() together with getAllIds() for class Magento\Catalog\Model\ResourceModel\Product\Attribute\Collection

- Fix unit test

nikshostko committed:
Merge pull request #55 from magento-epam/EPAM-MAGETWO-91620

MAGETWO-91620: It is not possible to use function setInAllAttributeSe…

nikshostko committed:
Merge pull request #56 from magento-epam/2.3-develop

Merge 2.3-develop to EPAM-PR-3

thomas-kl1 committed:
add declare strict type tag

Commit:
Fix exception message to avoid coupling of stock management with catalog

Commit:
Fixture correction to have correctly loaded invoice for creditmemo

magento-engcom-team committed:
ENGCOM-2810: Change error message about low stock #17642

 - Merge Pull Request magento/magento2#17642 from vkublytskyi/magento2:msi-e-msgs
 - Merged commits:
   1. c0cc6e10337dc519403f6e74b6d881619f36c0e2
   2. bd1653369332544465452a4f261bc81c9ccc9e95

magento-engcom-team committed:
ENGCOM-2328: [Forwardport-2.3] Fix Translation of error message on cart for deleted bundle option. #16778

 - Merge Pull Request magento/magento2#16778 from swnsma/magento2-1:Forwardport-2.3-Translation-Error-Message-On-Cart-For-Deleted-Bundle-Option
 - Merged commits:
   1. 0abbdd8b79d63b13be06340fe4e146adb45c9695
   2. a0329e44c7a5aeb4729ba0636cb0db9c08b5c5d9

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-93963' into 2.3-develop

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-73774' into 2.3-develop

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-93286' into 2.3-develop

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-93692' into 2.3-develop

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.3-develop

Accepted Public Pull Requests:
 - magento/graphql-ce#127: Fixed return value for getGraphQlClient in API-functional tests (by @rogyar)
 - magento/magento2#17572: [Forwardport] Catalog: Add unit tests for Cron classes (by @jignesh-baldha)
 - magento/magento2#17573: [Forwardport] updating lib LESS docs (by @jignesh-baldha)
 - magento/magento2#17571: [Forwardport] Refactor: remove some code duplication (by @arnoudhgz)
 - magento/magento2#17562: Fix wrong return type in StockRegistryInterface API (by @vasilii-b)
 - magento/magento2#17502: [Forwardport] Doesn't work if use date as condition for Catalog Price Rules (by @mage2pratik)
 - magento/magento2#17551: [Forwardport] Fix possible undefined index when caching config data (by @mage2pratik)
 - magento/magento2#17554: [Forwardport] Fix #10687 - Product image roles disappearing (by @eduard13)
 - magento/magento2#16784: [Forwardport] Fix the issue with "Shipping address is not set" exception (by @dmytro-ch)
 - magento/magento2#17499: [Forwardport] Braintree: test coverage (by @ronak2ram)
 - magento/magento2#17610: [Forwardport] Link logo in web setup wizard to back-end base URL (by @arnoudhgz)
 - magento/magento2#17604: [Forwardport] Fix for #15041 Adding a new fieldset to the admin category editor changes the position of the 'General' fieldset (by @jignesh-baldha)
 - magento/magento2#17603: [Forwardport] Braintree: Add unit test for CreditCard/TokenFormatter (by @jignesh-baldha)
 - magento/magento2#17596: Typo in Magento\Cms\Model\Wysiwyg\Images\Storage function resizeFile() (by @eduard13)


Fixed GitHub Issues:
 - magento/magento2#15085: StockRegistryInterface :: getLowStockItems() returns StockStatusCollection instead of StockItemCollection (reported by @jesse-dev) has been fixed in magento/magento2#17562 by @vasilii-b in 2.3-develop branch
   Related commits:
     1. 897e7df9441d121af626cace56cbc2a53e35c38d

 - magento/magento2#10687: Product image roles randomly disappear (reported by @boxyman) has been fixed in magento/magento2#17554 by @eduard13 in 2.3-develop branch
   Related commits:
     1. 1c74f5576011cd442dc086e52935c9fe68e7aa3e

 - magento/magento2#16555: "Shipping address is not set" exception in Multishipping Checkout. (reported by @p-bystritsky) has been fixed in magento/magento2#16784 by @dmytro-ch in 2.3-develop branch
   Related commits:
     1. ab1324ecc1d5e2990acfd4f14c7b77765d343a9e
     2. 7187c2e318e41e0e5bf0c93f663e17d543ee5380
     3. 9868428966220ddc23fe04f3af985eab2d66ff59
     4. 2c546c4d67d1b8434405874f80c02eb898c3dced

 - magento/magento2#13948: Sidebar shortcut to admin dashboard (Magento logo on top left) has no link in web setup wizard (reported by @doctormess) has been fixed in magento/magento2#17610 by @arnoudhgz in 2.3-develop branch
   Related commits:
     1. 43bbdb5730adbe56d161e38e236035c4af0a2c2f

 - magento/magento2#15041: Adding a new fieldset to the admin category editor changes the position of the 'General' fieldset. (reported by @leoquijano) has been fixed in magento/magento2#17604 by @jignesh-baldha in 2.3-develop branch
   Related commits:
     1. b6df8e7c9046b0125fb3211217572dd26c7cbee7

 - magento/magento2#17587: Typo in Magento\Cms\Model\Wysiwyg\Images\Storage function resizeFile($source, $keepRation = true) (reported by @simonworkhouse) has been fixed in magento/magento2#17596 by @eduard13 in 2.3-develop branch
   Related commits:
     1. 9a5bfd3f38f09f94553ef79da140291943a5e57a

Commit:
merge magento/2.3-develop into magento-tsg-csl3/2.3-develop-pr1

serhii-balko committed:
Merge remote-tracking branch 'origin/2.3-develop' into 2.3-develop-pr1

magento-engcom-team committed:
ENGCOM-2800: [Forwardport] Introduce Block Config Source #17598

 - Merge Pull Request magento/magento2#17598 from thomas-blackbird/magento2:2.3-develop-PR-port-16021
 - Merged commits:
   1. e735178bdfeaa1eb5071361fca9feac5aaa52047
   2. b00d4b274f0e4030cd97c9b9f961380fe159f17d
   3. 7563605f8ab69ea09b5f017e8472670842f4bfc9

davgrigoryan1992 committed:
MAGETWO-91701: Newsletter subscription is not correctly updated when user is registered on 2 stores

- Updated automated test. Updated ActionGroup names.

nikshostko committed:
Merge pull request #57 from magento-epam/MAGETWO-91701-V2

MAGETWO-91701: Newsletter subscription is not correctly updated when …

nikshostko committed:
Merge pull request #58 from magento-epam/2.3-develop

Merge 2.3-develop to EPAM-PR-2

tiagosampaio committed:
Replacing deprecated methods for tests.

Commit:
GraphQL-122: API-functional tests: assertResponseFields moved to the abstract class

tiagosampaio committed:
Replacing deprecated methods for Magento_Email module.

dthampy committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-90539

magento-engcom-team committed:
ENGCOM-2637: Corrected field descriptions #138

 - Merge Pull Request magento/graphql-ce#138 from keharper/graphql-ce:issue-136
 - Merged commits:
   1. 3623e5cbfff74b1ed9dd48c2c383adf87656eb0e

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into BugFixPR

Commit:
Merge branch 'BugFixPR' of https://github.com/magento-borg/magento2ce into BugFixPR

magento-engcom-team committed:
Merge pull request https://github.com/magento/magento2/pull/17646 to https://github.com/magento/magento2/pull/17645

magento-engcom-team committed:
ENGCOM-2422: API-functional tests: assertResponseFields moved to the abstract class #122

 - Merge Pull Request magento/graphql-ce#122 from magento/graphql-ce:assert-response-field-refactor
 - Merged commits:
   1. 33a06d3a2a5d0d06410941d8f9cc1b1c43195435
   2. 16c17baaeda0b48f9a7f3f13480f5325b2d205e9
   3. cbd470b14a538114c43abff9d9a8f0317632a25c

cpartica committed:
MAGETWO-92279: An incorrect result of declaration:generate:whitelist execution

- make generator exclude tables from app/etc/db_schema.xml
- fix unit tests
- create setup integration test for declaration:generate:whitelist

cpartica committed:
MAGETWO-92279: An incorrect result of declaration:generate:whitelist execution

- modifying db_schema_whitelist.json for ce

paliarush committed:
MAGETWO-94207: Cart GET for customer in REST returns 400 when cart not created

cpartica committed:
Merge remote-tracking branch 'mainline/2.3-develop' into MAGETWO-90531-WYSIWYG-special-char-Toolbar-Configuration

Commit:
Merge pull request #3035 from magento-borg/BugFixPR

[2.3.0-Regression] Bug Fixes PR

Commit:
merge magento/2.3-develop into magento-tsg-csl3/2.3-develop-pr1

serhii-balko committed:
Merge remote-tracking branch 'mainline/2.3-develop' into 2.3-develop

serhii-balko committed:
Merge branch '2.3-develop' into 2.3-develop-pr1

Commit:
ENGCOM-2810: Change error message about low stock #17642

Commit:
ENGCOM-2801: [Forwardport] Fix type error in Cart/Totals. #17620

Commit:
ENGCOM-2800: [Forwardport] Introduce Block Config Source #17598

Commit:
ENGCOM-2793: [Forwardport] 6305 - Resolved product custom option title save issue #17607

Commit:
ENGCOM-2788: Improve code quality subscriber new action #17232

Commit:
ENGCOM-2782: [Forwardport] Solution for User role issue with customer group #17574

Commit:
ENGCOM-2637: Corrected field descriptions #138

nikshostko committed:
Merge pull request #59 from magento-epam/2.3-develop

Merge 2.3-develop to EPAM-PR-3

davgrigoryan1992 committed:
MAGETWO-66666: Adding a product to cart from category page with an expired session does not allow product to be added

- Updated automated test. Renamed Action Group.

magento-engcom-team committed:
ENGCOM-2813: Replacing deprecated methods for tests. #17645

 - Merge Pull Request magento/magento2#17645 from tiagosampaio/magento2:2.3-develop-017
 - Merged commits:
   1. b37a9896b359279ebe96a67eec06f62264f68a03
   2. 432975dfc948619f8d38be7ed35aed140fb62fd4
   3. 470c2471455030287f9f7e5e90add84bd1880909

nikshostko committed:
Merge pull request #60 from magento-epam/SPRINT-1-MAGETWO-66666

MAGETWO-66666: Adding a product to cart from category page with an ex…

davgrigoryan1992 committed:
MAGETWO-91701: Newsletter subscription is not correctly updated when user is registered on 2 stores

- Updated automated test. Using Search Filter for deleting created customer.

nikshostko committed:
Merge pull request #61 from magento-epam/MAGETWO-91701-V2

MAGETWO-91701: Newsletter subscription is not correctly updated when …

nikshostko committed:
Merge pull request #62 from magento-epam/2.3-develop

Merge 2.3-develop to EPAM-PR-2

nikshostko committed:
Merge pull request #63 from magento-epam/MAGETWO-62891

Magetwo 62891

nikshostko committed:
Merge pull request #64 from magento-epam/MAGETWO-91493

MAGETWO-91493: MDC Framework Issues. Message passed as a service to T…

Commit:
Fix #17579 Typo in Gallery.php

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into 2.3-develop-pr32

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-93961' into 2.3-develop-pr3

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-93057' into 2.3-develop-pr3

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-93167' into 2.3-develop-pr3

serhii-balko committed:
Merge remote-tracking branch 'origin/MAGETWO-91682' into 2.3-develop-pr3

magento-engcom-team committed:
ENGCOM-2816: Fix #17579 Typo in Gallery.php #17659

 - Merge Pull Request magento/magento2#17659 from Spaggel/magento2:2.3-develop
 - Merged commits:
   1. 3a942809cf3f73114b5236e4f78bf2adf71411ca

cpartica committed:
MAGETWO-90531: WYSIWYG shows special character button in toolbar on product page - when test module is enabled

- moving location to make path shorter

davgrigoryan1992 committed:
MAGETWO-66666: Adding a product to cart from category page with an expired session does not allow product to be added

- Updated automated test.

cpartica committed:
MAGETWO-92279: An incorrect result of declaration:generate:whitelist execution

- simplifying code

Commit:
merge magento/2.3-develop into magento-honey-badgers/MAGETWO-92279-incorrect-result-generate-regenerate

nikshostko committed:
Merge pull request #67 from magento-epam/SPRINT-1-MAGETWO-66666

Sprint 1 magetwo 66666

davgrigoryan1992 committed:
MAGETWO-62891: New address is not marked as "Default Billing"

- Update automated test

davgrigoryan1992 committed:
MAGETWO-62891: New address is not marked as "Default Billing"

- Update automated test

YuliyaLabudova committed:
MAGETWO-91493: MDC Framework Issues. Message passed as a service to TransportBuilder

- Fix static and unit tests

nikshostko committed:
Merge pull request #68 from magento-epam/MAGETWO-62891

MAGETWO-62891: New address is not marked as "Default Billing"

nikshostko committed:
Merge pull request #69 from magento-epam/MAGETWO-91493

MAGETWO-91493: MDC Framework Issues. Message passed as a service to T…

Commit:
merge magento/2.3-develop into magento-architects/MAGETWO-94207-REST-400

cpartica committed:
MAGETWO-90531: WYSIWYG shows special character button in toolbar on product page - when test module is enabled

- removing invalid xml

cpartica committed:
MAGETWO-92279: An incorrect result of declaration:generate:whitelist execution

- add strict

magento-engcom-team committed:
ENGCOM-2818: Removed commented code #17346

 - Merge Pull Request magento/magento2#17346 from mage2pratik/magento2:remove-commented-code-6
 - Merged commits:
   1. 6c88c5891867b1bb8f4bfb983d00ca50d41182a6
   2. 01c3c71933a318f2e572961fdfb3d1fc2b1a5dd9
   3. f833fdab77cacef9b734e424c8301ebcc6061c16

Commit:
Merge pull request #3032 from magento-tsg-csl3/2.3-develop-pr1

[TSG-CSL3] Upporting 2.3 (pr1)

cpartica committed:
MAGETWO-92279: An incorrect result of declaration:generate:whitelist execution

- remove redundant strict types

Commit:
Merge branch '2.3-develop' of github.com:magento/magento2ce into EPAM-PR-3

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.3-develop

 - merged latest code from mainline branch

Karlasa committed:
remove empty abs classes

Karlasa committed:
removed empty abs class extend

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into MAGETWO-90531-WYSIWYG-special-char-Toolbar-Configuration

mage2pratik committed:
Fixed a couple of spelling mistakes

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into MAGETWO-90539

paliarush committed:
MAGETWO-94207: Cart GET for customer in REST returns 400 when cart not created

mage2pratik committed:
Replace strval() function by using direct type casting to (string)

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into MAGETWO-90863

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into MAGETWO-90591

rganin committed:
Merge pull request #3048 from magento-epam/EPAM-PR-3

[epam] MAGETWO-91620: It is not possible to use function setInAllAttributeSetsFilter() together with getAllIds() for class \Magento\Catalog\Model\ResourceModel\Product\Attribute\Collection

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.3-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
Merge pull request https://github.com/magento/magento2/pull/17671 to https://github.com/magento/magento2/pull/17672

magento-engcom-team committed:
ENGCOM-2820: [Forwardport] Added unit test for newsletter problem model #17666

 - Merge Pull Request magento/magento2#17666 from jignesh-baldha/magento2:2.3-develop-PR-port-17633
 - Merged commits:
   1. 52d3188c7e14ddb81bdb870d6d22da28d004d24a
   2. 65ad339d119619b04e37d92ff0d2f69579f9bb2d

magento-engcom-team committed:
ENGCOM-2821: [2.3] removed unused abstract class #17669

 - Merge Pull Request magento/magento2#17669 from Karlasa/magento2:less-extends
 - Merged commits:
   1. 9a7fcaa54faf763d428bd629f6b22fe1dd25ec44
   2. f7e93677a2888d321e34a0f2b74501f6beebda50

magento-engcom-team committed:
ENGCOM-2822: Replace strval() function by using direct type casting to (string) #17672

 - Merge Pull Request magento/magento2#17672 from mage2pratik/magento2:forwardport-patch-16849
 - Merged commits:
   1. bac945cd6da6ca152e8cb49a4cee69dac752681b
   2. 71104c498f3153159b3e762f8f139b1968f6b8a2
   3. 066f5892eeb95d96ec0f844a773ca7aa6cdeffca

dmytro-ch committed:
CMS: Add unit test for block model class

dmytro-ch committed:
Make "time12h" validation rule to be compatible with minify js

rogyar committed:
Integration test for reviews delete observer

magento-engcom-team committed:
ENGCOM-2824: [Forwardport] Update time12h javascript validation rule to be compatible with js minify #17688

 - Merge Pull Request magento/magento2#17688 from dmytro-ch/magento2:2.3-develop-PR-port-17652
 - Merged commits:
   1. fe088901a8e2c32fc75baca5b32ec332a022651c

magento-engcom-team committed:
ENGCOM-2826: [Forwardport] Fix Custom Attribute Group can not translate in catalog/product page #17683

 - Merge Pull Request magento/magento2#17683 from dmytro-ch/magento2:2.3-develop-PR-port-17602
 - Merged commits:
   1. aa465d5feba68ea657b61925376d628a1db27c5c

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.3-develop

Accepted Public Pull Requests:
 - magento/magento2#17642: Change error message about low stock (by @vkublytskyi)
 - magento/magento2#17620: [Forwardport] Fix type error in Cart/Totals. (by @nmalevanec)
 - magento/magento2#17598: [Forwardport] Introduce Block Config Source (by @thomas-blackbird)
 - magento/magento2#17607: [Forwardport] 6305 - Resolved product custom option title save issue (by @jignesh-baldha)
 - magento/magento2#17232: Improve code quality subscriber new action (by @arnoudhgz)
 - magento/magento2#17574: [Forwardport] Solution for User role issue with customer group (by @jignesh-baldha)
 - magento/graphql-ce#138: Corrected field descriptions (by @keharper)


Fixed GitHub Issues:
 - magento/magento2#12819: CartTotalRepository cannot handle extension attributes in quote addresses in 2.2.2 (reported by @marius-bica) has been fixed in magento/magento2#17620 by @nmalevanec in 2.3-develop branch
   Related commits:
     1. 370dc8492dc2b00d21e74dccf24e0abd6cca6d97

 - magento/magento2#12993: Type error in Cart/Totals (reported by @andrewhowdencom) has been fixed in magento/magento2#17620 by @nmalevanec in 2.3-develop branch
   Related commits:
     1. 370dc8492dc2b00d21e74dccf24e0abd6cca6d97

 - magento/magento2#6305: Can't save Customizable options (reported by @bachlee89) has been fixed in magento/magento2#17607 by @jignesh-baldha in 2.3-develop branch
   Related commits:
     1. d60b3cfbd8172afcccadcdef5da1774bf8628d36
     2. 71ce3fa2ee3487c6845286340c35937ec5083b47
     3. 170d02962d126e314980bd2382efee3c5ffa653b
     4. 6aebfe8c6007a71a5b2601e8050c1405b2685231

 - magento/magento2#16499: User role issue with customer group (reported by @emiprotech) has been fixed in magento/magento2#17574 by @jignesh-baldha in 2.3-develop branch
   Related commits:
     1. 0d7dc6cb6c4fb25839b73fcbf861a09a646379b3

nikshostko committed:
MAGETWO-66666: Adding a product to cart from category page with an expired session does not allow product to be added

- Updated automated test.

nikshostko committed:
Merge pull request #76 from magento-epam/SPRINT-1-MAGETWO-66666

MAGETWO-66666: Adding a product to cart from category page with an ex…

nikshostko committed:
Merge pull request #77 from magento-epam/2.3-develop

Merge 2.3-develop to EPAM-PR-2

magento-engcom-team committed:
ENGCOM-2829: [Forwardport] Translated validation error messages #17694

 - Merge Pull Request magento/magento2#17694 from osrecio/magento2:2.3-develop-PR-port-17575
 - Merged commits:
   1. b27e02824049ac4151b1ec330a8494bdb98bcc3e

serhii-balko committed:
Merge branch '2.3-develop' into 2.3-develop-pr3

Commit:
Fixed incorrect namespace

dmytro-ch committed:
Message component fix: the message type is always error when parameters specified

Commit:
Merge remote-tracking branch 'origin/2.3-develop' into 87-Fetch-attribute-values-and-labels-for-customAttributeMetadata

Commit:
GraphQL-87: Fetch attribute values and labels for customAttributeMetadata

Commit:
GraphQL-87: Fetch attribute values and labels for customAttributeMetadata

Commit:
GraphQL-87: Fetch attribute values and labels for customAttributeMetadata

Commit:
Review: add missing unit test for Observer

magento-engcom-team committed:
ENGCOM-2811: Exclude already minified js files from minification in the Swagger mo… #17627

 - Merge Pull Request magento/magento2#17627 from hostep/magento2:fix-issue-16927-for-magento-23
 - Merged commits:
   1. 6aeec4710d0decd5161bb3855e83540df81e6a14

magento-engcom-team committed:
ENGCOM-2836: [Forwardport] CMS: Add missing unit tests for model classes #17681

 - Merge Pull Request magento/magento2#17681 from dmytro-ch/magento2:2.3-unit-test-cms-model-block
 - Merged commits:
   1. 26fd77357014bb9a4359ebdaa53e8ae5819d10b1

magento-engcom-team committed:
ENGCOM-2839: [Forwardport] Message list component fix: the message type is always error when parameters specified #17703

 - Merge Pull Request magento/magento2#17703 from dmytro-ch/magento2:fix/2.3-message-type-is-always-error
 - Merged commits:
   1. 88aa0bd9b77856f2f10a507cb120fd4e9309fa96

Commit:
merge magento/2.3-develop into magento-tsg/2.3-develop-pr32

magento-engcom-team committed:
ENGCOM-2840: 87 Fetch attribute values and labels for customAttributeMetadata #108

 - Merge Pull Request magento/graphql-ce#108 from magento/graphql-ce:87-Fetch-attribute-values-and-labels-for-customAttributeMetadata
 - Merged commits:
   1. 7674eb1713c7f348cd7fc8b8bc3c7643b49a3638
   2. 4aa808e537bf7526d81a3bc18910ad2ddcc8400f
   3. 3af924b5731b3d0894f6b948bf1dabae8a7f8e84
   4. a9361b9b1c8df62246112263ee066c02a43734bd
   5. b77f96e83998e3b305390ef0ba0902b7a2d90211
   6. a84e6d85fd0cb6f2b9c8541a2cefc5a2d087aef9

Commit:
MAGETWO-90591: Add Selected Button shouldn't be enabled if there are no images available

- update function test

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into MAGETWO-90531-WYSIWYG-special-char-Toolbar-Configuration

Commit:
Merge branch 'MAGETWO-90863' into MAGETWO-90531-WYSIWYG-special-char-Toolbar-Configuration

Commit:
Merge branch 'MAGETWO-90539' into MAGETWO-90531-WYSIWYG-special-char-Toolbar-Configuration

dmytro-ch committed:
Add unit test for Sales Rule model class

Commit:
merge magento/2.3-develop into magento-architects/MAGETWO-94207-REST-400

rganin committed:
Merge branch '2.3-develop' of github.com:magento/magento2ce into ENGCOM-2088

rganin committed:
ENGCOM-2841: Fix external Sodium Encryption PR

rganin committed:
ENGCOM-2841: Fix external Sodium Encryption PR

rganin committed:
ENGCOM-2841: Fix external Sodium Encryption PR

rganin committed:
ENGCOM-2841: Fix external Sodium Encryption PR

Commit:
Merge pull request #3060 from magento-borg/BugFixPR

[2.3.0-Regression] Bug Fixes PR

rganin committed:
Merge branch '2.3-develop' of github.com:magento/magento2ce into ENGCOM-2088

Commit:
Review: fixing the properties' naming

serhii-balko committed:
Merge remote-tracking branch 'origin/2.3-develop' into 2.3-develop-pr3

yogeshsuhagiya committed:
Translated menu titles

magento-engcom-team committed:
ENGCOM-2846: [Forwardport] Translated menu titles #17713

 - Merge Pull Request magento/magento2#17713 from nmalevanec/magento2:2.3-develop-PR-port-17521
 - Merged commits:
   1. 81edc360f082d662f3b298e28d85d79bdd9844b4

ananth-iyer committed:
Fixed review list ajax if product not exist redirect to 404 page

magento-engcom-team committed:
ENGCOM-2849: [Forwardport] Integration test for reviews delete observer #17712

 - Merge Pull Request magento/magento2#17712 from nmalevanec/magento2:2.3-develop-PR-port-17690
 - Merged commits:
   1. 704329e1e963fa6e880c9fd20e89e5888e876681
   2. 5a03f97fa113e20901a2d4d8d104cc83a3c2ab87

Commit:
Merge pull request #3037 from magento-tsg-csl3/2.3-develop-pr3

[TSG-CSL3] Upporting 2.3 (pr3)

Commit:
merge magento/2.3-develop into magento-honey-badgers/MAGETWO-92279-incorrect-result-generate-regenerate

cpartica committed:
Merge pull request #3045 from magento-honey-badgers/MAGETWO-92279-incorrect-result-generate-regenerate

[honey] MAGETWO-92279: An incorrect result of declaration:generate:whitelist execution

Commit:
merge magento/2.3-develop into magento-architects/MAGETWO-94207-REST-400

cpartica committed:
Merge pull request #3046 from magento-architects/MAGETWO-94207-REST-400

[architects] MAGETWO-94207: Cart GET for customer in REST returns 400 when cart not created

magento-engcom-team committed:
ENGCOM-2852: [Forwardport] magento/magento2#12250: View.xml is inheriting image sizes from paren… #17725

 - Merge Pull Request magento/magento2#17725 from nmalevanec/magento2:2.3-develop-PR-port-14537
 - Merged commits:
   1. d6c7459b26ce2e4e24ef2bbf8547e2790c01324c
   2. f2a1c08a401a338fa93f6f35aea34a9643ae811e
   3. 240e18df3e108c5bb244ce01945f8e9415675fb4

magento-engcom-team committed:
ENGCOM-2845: [Forwardport] Review: Adding missing unit test for Observer #17704

 - Merge Pull Request magento/magento2#17704 from eduard13/magento2:2.3-develop-process-product-after-delete-event-ut
 - Merged commits:
   1. 7220e131f9ae28ad599197dbaf6a0443dc8e70ca
   2. 5b568abe49ff56f160fdc2c3572f5a2290aa47ca

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into 2.3-develop-pr32

magento-engcom-team committed:
ENGCOM-2856: [Forwardport] Fix sending duplicate emails #17714

 - Merge Pull Request magento/magento2#17714 from nmalevanec/magento2:2.3-develop-PR-port-17484
 - Merged commits:
   1. d7ab99d8ba8914ffc9e47e396cda238c825712ce
   2. 133311494ebc8bd60df6d1a1de4955aee843b860

magento-engcom-team committed:
ENGCOM-2857: [Forwardport] Add unit test for Sales Rule model class #17717

 - Merge Pull Request magento/magento2#17717 from nmalevanec/magento2:2.3-develop-PR-port-17710
 - Merged commits:
   1. c43443b0e5e6493c8b59ee857c3c1c56ce993684

magento-engcom-team committed:
ENGCOM-2858: [Forwardport] Fixed review list ajax if product not exist redirect to 404 page #17721

 - Merge Pull Request magento/magento2#17721 from nmalevanec/magento2:2.3-develop-PR-port-15369
 - Merged commits:
   1. 3f0168cbf6589e850cbc4ca4b6ca3796a04330ab

davgrigoryan1992 committed:
MAGETWO-66666: Adding a product to cart from category page with an expired session does not allow product to be added

- Updated automated test.

magento-engcom-team committed:
ENGCOM-2862: [Forwardport] Magento PayPal checkout fails if email sending fails / other payment does not #17723

 - Merge Pull Request magento/magento2#17723 from nmalevanec/magento2:2.3-develop-PR-port-13133
 - Merged commits:
   1. 57f9fef74874097b2a7175421a8456dd618396f1

jissereitsma committed:
Add RETURN_SUCCESS return

magento-engcom-team committed:
ENGCOM-2864: [Forwardport] Enhancements to module:status command #17741

 - Merge Pull Request magento/magento2#17741 from nmalevanec/magento2:2.3-develop-PR-port-15543
 - Merged commits:
   1. 934fedc7234d5036b4a9672014486c0d0be8cc4a
   2. 0198f967d0ba130d28380d28817689fb08724630
   3. 3c74589177d57f29b3dc528a830c8c49c546b18b
   4. 245d4efc4c0ccd0b0fe1de4c099634d01c742869

sanganinamrata committed:
Removed double occurrence of 'it' from sentences.

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into MAGETWO-90591

sunilit42 committed:
Misleading data-container in product list

vprohorov committed:
MAGETWO-91701: Newsletter subscription is not correctly updated when user is registered on 2 stores

- Changing customer account getSubscriptionObject behavior

mage2pratik committed:
Fixed a couple of spelling mistakes

rganin committed:
Merge branch '2.3-develop' of github.com:magento/magento2ce into ENGCOM-2088

magento-engcom-team committed:
ENGCOM-2865: [Forwardport] Fix typo in DeploymentConfig class #17747

 - Merge Pull Request magento/magento2#17747 from nmalevanec/magento2:2.3-develop-PR-port-15293
 - Merged commits:
   1. fda50209019d744970f839b08cd48fdc675bb375

magento-engcom-team committed:
ENGCOM-2866: [Forwardport] Fixed a couple of spelling mistakes #17755

 - Merge Pull Request magento/magento2#17755 from mage2pratik/magento2:forwardport-patch-16980
 - Merged commits:
   1. a3c361a25d2747971e728c231df1026e29e93b0d

magento-engcom-team committed:
ENGCOM-2867: [Forwardport] Correct spelling mistakes in Model and library files. #17743

 - Merge Pull Request magento/magento2#17743 from nmalevanec/magento2:2.3-develop-PR-port-16230
 - Merged commits:
   1. 1589c7985b691c00e75c21a7ebf705391afd511e

magento-engcom-team committed:
ENGCOM-2868: [Forwardport] Removed double occurrence of 'it' from sentences. #17742

 - Merge Pull Request magento/magento2#17742 from nmalevanec/magento2:2.3-develop-PR-port-16240
 - Merged commits:
   1. b92eefb6b13fc38982ac438c6aec6021dcd56810

magento-engcom-team committed:
ENGCOM-2869: [Forwardport] Misleading data-container in product list #17751

 - Merge Pull Request magento/magento2#17751 from nmalevanec/magento2:2.3-develop-PR-port-15350
 - Merged commits:
   1. cba6c39197568d8255541b9871e53582d72f1b60

nikshostko committed:
MAGETWO-91701: Newsletter subscription is not correctly updated when user is registered on 2 stores

- Update functional test

nikshostko committed:
Merge pull request #78 from magento-epam/MAGETWO-91701-V2

Magetwo 91701 v2

nikshostko committed:
Merge pull request #79 from magento-epam/SPRINT-1-MAGETWO-66666

MAGETWO-66666: Adding a product to cart from category page with an ex…

nikshostko committed:
Merge pull request #80 from magento-epam/2.3-develop

Merge 2.3-develop to EPAM-PR-2

Commit:
Merge pull request #3043 from magento-tsg/2.3-develop-pr32

[TSG] Upporting for 2.3 (pr32) (2.3.0)

rganin committed:
Merge branch '2.3-develop' of github.com:magento/magento2ce into ENGCOM-2088

rganin committed:
Merge pull request #3072 from magento-earl-grey/ENGCOM-2088

[earl] ENGCOM-2088: Sodium Encryption #135

nikshostko committed:
Merge pull request #81 from magento-epam/MAGETWO-62891

MAGETWO-62891: New address is not marked as "Default Billing"

nikshostko committed:
Merge pull request #82 from magento-epam/2.3-develop

Merge 2.3-develop to EPAM-PR-4

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into MAGETWO-90591

Commit:
Merge pull request #3079 from magento-borg/MAGETWO-90591

[borg] MAGETWO-90591: Add Selected Button shouldn't be enabled if there are no images available

nikshostko committed:
Merge pull request #83 from magento-epam/2.3-develop

Merge 2.3-develop to EPAM-PR-2

Commit:
ENGCOM-2868: [Forwardport] Removed double occurrence of 'it' from sentences. #17742

Commit:
ENGCOM-2867: [Forwardport] Correct spelling mistakes in Model and library files. #17743

Commit:
ENGCOM-2866: [Forwardport] Fixed a couple of spelling mistakes #17755

Commit:
ENGCOM-2857: [Forwardport] Add unit test for Sales Rule model class #17717

Commit:
ENGCOM-2852: [Forwardport] magento/magento2#12250: View.xml is inheriting image sizes from paren… #17725

Commit:
ENGCOM-2849: [Forwardport] Integration test for reviews delete observer #17712

Commit:
ENGCOM-2846: [Forwardport] Translated menu titles #17713

Commit:
ENGCOM-2845: [Forwardport] Review: Adding missing unit test for Observer #17704

Commit:
ENGCOM-2840: 87 Fetch attribute values and labels for customAttributeMetadata #108

Commit:
ENGCOM-2836: [Forwardport] CMS: Add missing unit tests for model classes #17681

Commit:
ENGCOM-2829: [Forwardport] Translated validation error messages #17694

Commit:
ENGCOM-2826: [Forwardport] Fix Custom Attribute Group can not translate in catalog/product page #17683

Commit:
ENGCOM-2824: [Forwardport] Update time12h javascript validation rule to be compatible with js minify #17688

Commit:
ENGCOM-2822: Replace strval() function by using direct type casting to (string) #17672

Commit:
ENGCOM-2821: [2.3] removed unused abstract class #17669

Commit:
ENGCOM-2820: [Forwardport] Added unit test for newsletter problem model #17666

Commit:
ENGCOM-2818: Removed commented code #17346

Commit:
ENGCOM-2816: Fix #17579 Typo in Gallery.php #17659

Commit:
ENGCOM-2813: Replacing deprecated methods for tests. #17645

Commit:
ENGCOM-2811: Exclude already minified js files from minification in the Swagger mo… #17627

Commit:
ENGCOM-2422: API-functional tests: assertResponseFields moved to the abstract class #122

Commit:
ENGCOM-2328: [Forwardport-2.3] Fix Translation of error message on cart for deleted bundle option. #16778

rganin committed:
Merge branch '2.3-develop' of github.com:magento/magento2ce into EPAM-PR-4

Commit:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.3-develop

Accepted Public Pull Requests:
 - magento/magento2#17742: [Forwardport] Removed double occurrence of 'it' from sentences. (by @nmalevanec)
 - magento/magento2#17743: [Forwardport] Correct spelling mistakes in Model and library files. (by @nmalevanec)
 - magento/magento2#17755: [Forwardport] Fixed a couple of spelling mistakes (by @mage2pratik)
 - magento/magento2#17717: [Forwardport] Add unit test for Sales Rule model class (by @nmalevanec)
 - magento/magento2#17725: [Forwardport] magento/magento2#12250: View.xml is inheriting image sizes from paren (by @nmalevanec)
 - magento/magento2#17712: [Forwardport] Integration test for reviews delete observer (by @nmalevanec)
 - magento/magento2#17713: [Forwardport] Translated menu titles (by @nmalevanec)
 - magento/magento2#17704: [Forwardport] Review: Adding missing unit test for Observer (by @eduard13)
 - magento/graphql-ce#108: 87 Fetch attribute values and labels for customAttributeMetadata (by @klimart)
 - magento/magento2#17681: [Forwardport] CMS: Add missing unit tests for model classes (by @dmytro-ch)
 - magento/magento2#17694: [Forwardport] Translated validation error messages (by @osrecio)
 - magento/magento2#17683: [Forwardport] Fix Custom Attribute Group can not translate in catalog/product page (by @dmytro-ch)
 - magento/magento2#17688: [Forwardport] Update time12h javascript validation rule to be compatible with js minify (by @dmytro-ch)
 - magento/magento2#17672: Replace strval() function by using direct type casting to (string) (by @mage2pratik)
 - magento/magento2#17669: [2.3] removed unused abstract class (by @Karlasa)
 - magento/magento2#17666: [Forwardport] Added unit test for newsletter problem model (by @jignesh-baldha)
 - magento/magento2#17346: Removed commented code (by @mage2pratik)
 - magento/magento2#17659: Fix #17579 Typo in Gallery.php (by @Spaggel)
 - magento/magento2#17645: Replacing deprecated methods for tests. (by @tiagosampaio)
 - magento/magento2#17627: Use '.min' in filenames of already minified js files in the Swagger module so they aren't getting minified again in production, fixes #16927 - for Magento 2.3 (by @hostep)
 - magento/graphql-ce#122: API-functional tests: assertResponseFields moved to the abstract class (by @rogyar)
 - magento/magento2#16778: [Forwardport-2.3] Fix Translation of error message on cart for deleted bundle option. (by @swnsma)


Fixed GitHub Issues:
 - magento/magento2#12250: View.xml is inheriting image sizes from parent (so an optional field is replaced by the value of parent) (reported by @quisse) has been fixed in magento/magento2#17725 by @nmalevanec in 2.3-develop branch
   Related commits:
     1. d6c7459b26ce2e4e24ef2bbf8547e2790c01324c
     2. f2a1c08a401a338fa93f6f35aea34a9643ae811e
     3. 240e18df3e108c5bb244ce01945f8e9415675fb4

 - magento/magento2#17648: UI validation rule for valid time am/pm doesn't work when js is minified (reported by @markoshust) has been fixed in magento/magento2#17688 by @dmytro-ch in 2.3-develop branch
   Related commits:
     1. fe088901a8e2c32fc75baca5b32ec332a022651c

 - magento/magento2#17579: Typo in vendor/magento/module-catalog/Block/Adminhtml/Product/Helper/Form/Gallery.php (reported by @adamkingfisher) has been fixed in magento/magento2#17659 by @Spaggel in 2.3-develop branch
   Related commits:
     1. 3a942809cf3f73114b5236e4f78bf2adf71411ca

 - magento/magento2#16927: 2.2.5 Swagger: With JS minification enabled, the swagger-ui-bundle.js becomes corrupted (reported by @Silarn) has been fixed in magento/magento2#17627 by @hostep in 2.3-develop branch
   Related commits:
     1. 6aeec4710d0decd5161bb3855e83540df81e6a14

Commit:
merge magento/2.3-develop into magento-epam/EPAM-PR-4

rganin committed:
Merge branch '2.3-develop' of github.com:magento/magento2ce into EPAM-PR-4

rganin committed:
Merge branch 'EPAM-PR-4' of github.com:magento-epam/magento2ce into EPAM-PR-4

Commit:
ENGCOM-2869: [Forwardport] Misleading data-container in product list #17751

Commit:
ENGCOM-2865: [Forwardport] Fix typo in DeploymentConfig class #17747

Commit:
ENGCOM-2864: [Forwardport] Enhancements to module:status command #17741

Commit:
ENGCOM-2862: [Forwardport] Magento PayPal checkout fails if email sending fails / other payment does not #17723

Commit:
ENGCOM-2858: [Forwardport] Fixed review list ajax if product not exist redirect to 404 page #17721

Commit:
ENGCOM-2856: [Forwardport] Fix sending duplicate emails #17714

Commit:
ENGCOM-2839: [Forwardport] Message list component fix: the message type is always error when parameters specified #17703

rganin committed:
Merge pull request #3081 from magento-epam/EPAM-PR-4

[epam] MAGETWO-62891: New address is not marked as "Default Billing"

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.3-develop

 - merged latest code from mainline branch

rganin committed:
Merge branch '2.3-develop' of github.com:magento/magento2ce into EPAM-PR-2

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.3-develop

Accepted Public Pull Requests:
 - magento/magento2#17751: [Forwardport] Misleading data-container in product list (by @nmalevanec)
 - magento/magento2#17747: [Forwardport] Fix typo in DeploymentConfig class (by @nmalevanec)
 - magento/magento2#17741: [Forwardport] Enhancements to module:status command (by @nmalevanec)
 - magento/magento2#17723: [Forwardport] Magento PayPal checkout fails if email sending fails / other payment does not (by @nmalevanec)
 - magento/magento2#17721: [Forwardport] Fixed review list ajax if product not exist redirect to 404 page (by @nmalevanec)
 - magento/magento2#17714: [Forwardport] Fix sending duplicate emails (by @nmalevanec)
 - magento/magento2#17703: [Forwardport] Message list component fix: the message type is always error when parameters specified (by @dmytro-ch)


Fixed GitHub Issues:
 - magento/magento2#15319: misleading data-container in product list (reported by @DanielRuf) has been fixed in magento/magento2#17751 by @nmalevanec in 2.3-develop branch
   Related commits:
     1. cba6c39197568d8255541b9871e53582d72f1b60

 - magento/magento2#13102: review/product/listAjax/id/{{non existent id}/ (reported by @ruthger92) has been fixed in magento/magento2#17721 by @nmalevanec in 2.3-develop branch
   Related commits:
     1. 3f0168cbf6589e850cbc4ca4b6ca3796a04330ab

 - magento/magento2#17700: Message list component: the message type is always error when parameters specified (reported by @dmytro-ch) has been fixed in magento/magento2#17703 by @dmytro-ch in 2.3-develop branch
   Related commits:
     1. 88aa0bd9b77856f2f10a507cb120fd4e9309fa96

rganin committed:
Merge branch '2.3-develop' of github.com:magento/magento2ce into EPAM-PR-2

rganin committed:
Merge pull request #3085 from magento-epam/EPAM-PR-2

[epam] MAGETWO-66666: Adding a product to cart from category page with an expired session does not allow product to be added

Commit:
Merge remote-tracking branch 'mainline/2.3-develop' into merge-mainline

# Conflicts:
#	composer.lock
#	dev/tests/integration/testsuite/Magento/CatalogSearch/Model/Search/FilterMapper/StockStatusFilterTest.php
#	dev/tests/integration/testsuite/Magento/Quote/Model/QuoteTest.php

Commit:
MSI: Merge mainline

-- Update composer lock

Commit:
MSI: Merge mainline

-- Quick fix of integration tests (https://github.com/magento-engcom/msi/issues/1586)

Commit:
MSI: Merge mainline

-- Skip of integration tests (https://github.com/magento-engcom/msi/issues/1587)
